viper/parser/parser_utils.py

```diff
@@ -238,9 +247,13 @@ def make_byte_array_copier(destination, source):
         raise TypeMismatchException("Cannot cast from greater max-length %d to shorter max-length %d" % (source.typ.maxlen, destination.typ.maxlen))
     # Special case: memory to memory
     if source.location == "memory" and destination.location == "memory":
-        return LLLnode.from_list(
+        gas_calculation = opcodes.GIDENTITYBASE + \
+            opcodes.GIDENTITYWORD * (ceil32(source.typ.maxlen) // 32)
+        o = LLLnode.from_list(
```

Comment:
I love how you're calculating the memory usage.

Reply:
:smile: thanks! I picked it from the pyethereum specials.py file. We can now do it in other known precompiled contract in the same way using `add_gas`.